Index: hex-a-hop.svn/gfx.cpp
===================================================================
--- hex-a-hop.svn.orig/gfx.cpp	2007-09-20 00:48:52.000000000 +0200
+++ hex-a-hop.svn/gfx.cpp	2007-09-20 00:49:05.000000000 +0200
@@ -197,7 +197,7 @@
 	SDL_Flip(screen);
 */
 
-	SDL_Init(SDL_INIT_EVERYTHING | SDL_INIT_NOPARACHUTE);
+	SDL_Init(SDL_INIT_VIDEO | SDL_INIT_NOPARACHUTE);
 
 	SDL_Surface* icon = SDL_LoadBMP("graphics/icon.bmp");
 	if (icon)
