Index: hex-a-hop.svn/hex_puzzzle.cpp
===================================================================
--- hex-a-hop.svn.orig/hex_puzzzle.cpp	2007-09-20 00:49:31.000000000 +0200
+++ hex-a-hop.svn/hex_puzzzle.cpp	2007-09-20 00:49:33.000000000 +0200
@@ -2790,10 +2790,6 @@
 			{
 				RenderTile(false, editTile, mousex+scrollX, mousey+scrollY);
 			}
-			else
-			{
-				Print(mousex, mousey-2, "\x7f");
-			}
 		}
 	}
 
@@ -4128,7 +4124,7 @@
 	//	};
 	//	static SDL_Cursor * c = SDL_CreateCursor(data, mask, 32, 32, 1, 1);
 	//	SDL_SetCursor(c);
-		SDL_ShowCursor(0);
+		SDL_ShowCursor(1);
 	}
 	void FreeGraphics()
 	{
