Index: hex-a-hop.svn/menus.h
===================================================================
--- hex-a-hop.svn.orig/menus.h	2007-09-20 00:49:10.000000000 +0200
+++ hex-a-hop.svn/menus.h	2007-09-20 00:49:17.000000000 +0200
@@ -125,7 +125,7 @@
 //Item 1 (22)
 /* TRANSLATORS: Normally you jump from one plate to another. The golden jump (a
    pickup) allows you to jump and land on the *same* plate */
-_("Collecting the golden jump pickups will allow you to  do a big vertical jump.    Try it out on different types of tile.    Use the space bar or return key to jump. Or click  on the tile you're currently on with the mouse."),
+_("Collecting the golden jump pickups will allow you to  do a big vertical jump.    Try it out on different types of tiles.    Use the space bar or return key to jump. Or click  on the tile you're currently on with the mouse."),
 
 0,0,
 
@@ -147,7 +147,7 @@
 /* TRANSLATORS: This string is copied twice into the POT file to workaround a
    gettext limitation (no macro expansion). The extracted string "Welcome to "
    will not be used. */
-_("Welcome to " GAMENAME "!    This is a puzzle game based on hexagonal tiles.  There is no time limit and no real-time elements, so  take as long as you like.    Use the cursor keys or click on the arrows to  scroll through the help pages. More pages will be  added as you progress through the game."),
+_("Welcome to " GAMENAME "!    This is a puzzle game based on hexagonal tiles.  There is no time limit and no real-time element, so  take as long as you like.    Use the cursor keys or click on the arrows to  scroll through the help pages. More pages will be  added as you progress through the game."),
 };
 
 struct HintMessage : public Menu
Index: hex-a-hop.svn/level_list.h
===================================================================
--- hex-a-hop.svn.orig/level_list.h	2007-09-20 00:49:10.000000000 +0200
+++ hex-a-hop.svn/level_list.h	2007-09-20 00:49:17.000000000 +0200
@@ -57,7 +57,7 @@
 {0,		"2005_11_19\\turntables.lev",				_("  Roundabouts"),},
 {0,		"2005_11_19\\two fish.lev",					_("  Two  Fish"),},
 {0,		"3_2hitfloor\\all wound up.lev",			_("All  Wound  Up"),},
-{0,		"3_2hitfloor\\collapse2.lev",				_("  Toughened   Tiles"),},
+{0,		"3_2hitfloor\\collapse2.lev",				_("  Toughened  Tiles"),},
 {0,		"3_2hitfloor\\Island.lev",					_("  Island"),},
 {0,		"3_2hitfloor\\more stripes.lev",			_("  More  Stripes"),},
 {0,		"3_2hitfloor\\Stripey.lev",					_("  Stripes"),},
